confgenerator/self_logs.go

```diff
@@ -113,14 +97,84 @@ func generateFluentBitSelfLogsComponents(ctx context.Context) []fluentbit.Compon
 			},
 		},
 	}.Components(ctx, fluentBitSelfLogsTag, "self-logs-severity")...)
-	out = append(out, fluentbit.TranslationComponents(fluentBitSelfLogsTag, "severity", severityKey, true,
-		[]struct{ SrcVal, DestVal string }{
-			{"debug", "DEBUG"},
-			{"error", "ERROR"},
-			{"info", "INFO"},
-			{"warn", "WARNING"},
-		})...,
-	)
+	out = append(out, LoggingProcessorModifyFields{
+		Fields: map[string]*ModifyField{
+			"severity": {
+				MoveFrom: "jsonPayload.severity",
+				MapValues: map[string]string{
+					"error": "ERROR",
+					"warn":  "WARNING",
+					"info":  "INFO",
+					"debug": "DEBUG",
+				},
+				MapValuesExclusive: true,
+			},
+		},
+	}.Components(ctx, fluentBitSelfLogsTag, "mapseverityvalues")...)
+	return out
+}
+
+type healthLogsMatch struct {
+	key     string
+	match   string
+	message string
+	code    string
+}
+
+var healthLogsMatchList = []healthLogsMatch{
+	{
+		key:     "message",
+		match:   `\[error\]\s\[lib\]\sbackend\sfailed`,
+		message: "LogggingPipelineStoppedWorking-PleaseVerifyYourOpsAgentConfig",
+		code:    "LogPipelineErr",
+	},
+	{
+		key:     "message",
+		match:   `\[error\]\s\[parser\]\scannot\sparse`,
+		message: "IncorrectParsingConfiguration-PleaseVerifyYourOpsAgentConfig",
+		code:    "LogParseErr",
+	},
+}
+
+func generateSampleSelfLogsComponents(ctx context.Context) []fluentbit.Component {
```

Comment:
nit: the generateSampleSelfLogsComponents name is a bit misleading. Maybe
generateMatchedSampledSelfLogsComponents or 
generateSampledSelfLogsComponents (just adding a D)

I think just generateSampleXXX gives the idea that we are creating an example, not that the logs themselves are sampled.

Reply:
Went for `generateSelfLogsSamplingComponents` . WDYT ?

Reply:
Sure, that works